File: tests/poll_hardware.cpp
#include <trusimd.hpp>
#include <iostream>

int main(int, char **argv) {
  trusimd_hardware *h;
  int n = trusimd_poll(&h);
  if (n == -1) {
    std::cerr << argv[0] << ": error: " << trusimd_strerror(trusimd_errno)
              << std::endl;
    return -1;
  }
  for (int i = 0; i < n; i++) {
    std::cout << h[i].description << std::endl;
  }
  return 0;
}

File: tests/poll_hardware.f90
program poll_hardware

  use trusimd

  type(trusimd_hardware), allocatable :: hardwares(:)
  
  if (trusimd_poll(hardwares) == -1) then
    print '(A)', trim(trusimd_strerror(trusimd_errno))
    stop 0
  end if

  do i = 1, size(hardwares)
    print '(A)', trim(hardwares(i)%description)
  end do

end program poll_hardware
